include/spdk/nvme.h

```diff
@@ -386,11 +386,11 @@ struct spdk_nvme_qpair;
  * Signature for the callback function invoked when a timeout is
  * detected on a request.
  */
-typedef void (*spdk_nvme_timeout_cb)(void *cb_arg,
-				     struct spdk_nvme_ctrlr *ctrlr,
+typedef void (*spdk_nvme_timeout_cb)(struct spdk_nvme_ctrlr *ctrlr,
 				     struct spdk_nvme_qpair *qpair,
+				     void *cb_arg,
```

Comment:
The timeout is extended by refreshing tr->submit_tick before calling the callback. If, for some reason, the abort does not clean up the command, this "second" timeout will indicate that something is wrong and that the controller should be reset.